dogfooding/src/main/kotlin/io/getstream/video/android/ui/DogfoodingNavHost.kt

```diff
@@ -38,6 +46,34 @@ fun AppNavHost(
     navController: NavHostController = rememberNavController(),
     startDestination: String = AppScreens.Login.destination,
 ) {
+    val context = LocalContext.current
+
+    val qrScannedCallback = remember {
+        OnSuccessListener<Barcode> {
+            val url = it.url?.url
+            val callId = if (url != null) {
+                val id = Uri.parse(url).getQueryParameter("id")
+                if (!id.isNullOrEmpty()) {
+                    id
+                } else {
+                    null
+                }
+            } else {
+                null
+            }
+
+            if (!callId.isNullOrEmpty()) {
+                context.startActivity(DeeplinkingActivity.createIntent(context, callId))
+            } else {
+                Toast.makeText(
+                    context,
+                    "Unrecognised meeting QR code format",
+                    Toast.LENGTH_SHORT,
+                ).show()
+            }
+        }
+    }
```

Comment:
I'm curious about the rationale behind exposing the `qrScannedCallback` within the `NavHost` rather than within the `CallJoinScreen`. It seems like we could simplify this by introducing a straightforward composable function named `rememberQRScannedCallback` and locating it within the `CallJoinScreen`.

Reply:
No rationale here :-). Good point. I changed it.